File: 141.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>

// Values below 100000 are 9, 10404, 16900, 97344
// sq roots are 3, 102, 130, 312 

int main(){

    int nmin = 1;
    int nmx = 100000;
    int q,r,n,d;
    float sq;
    std::vector<int> tpvec;
    std::vector<int> vls;
    std::vector<int> sqvls;
    
    for(n=nmin; n<nmx; n++)
    {
        
        if(n%10000==0)
            std::cout<<n<<std::endl;

        for(d = 1; d<n; d++)
        {

            tpvec.clear();
            q = n/d; r = n%d;
            tpvec.push_back(r); tpvec.push_back(d); tpvec.push_back(q);
            std::sort(tpvec.begin(),tpvec.end());

            if(tpvec[0]!=0 && float(tpvec[2])/tpvec[1] == float(tpvec[1])/tpvec[0])
            {
                vls.push_back(n);
            }
        }
    }

    for(int i = 0; i<vls.size(); i++)
    {
        sq = std::sqrt(vls[i]);
        if(std::abs(sq - int(sq))<1e-8)
            sqvls.push_back(vls[i]);
    }


    // for(int j = 0; j<tpvec.size(); ++j)
    //    std::cout << tpvec[j] << std::endl;


    for(int k = 0; k< sqvls.size(); ++k)
    {
        std::cout << sqvls[k] << std::endl;
    }

    return 0;
}

File: 141.py
import numpy as np
import math

vls = []

gdlst = [9, 10404, 16900, 97344]
sqs = [n*n for n in range(1,1000)]

nmx = 100
sm = 0


#for n in range(nmx):
for n in sqs:
    for d in range(1,n):
        q = n/d
        r = n%d
        vec = np.sort(np.array([d, q, r]))
#        vec = np.array([r,d,q])
        if(vec[0]!=0 and float(vec[2])/vec[1] == float(vec[1])/vec[0]):
            sm += n

#sqvls = []

#for num in vls:
#    sq = math.sqrt(num[0])
#     if(abs(sq - int(sq))<1e-8):
 #       sqvls.append(num)

print sm

